packages/client/src/components/LoginForm/LoginForm.tsx

```diff
@@ -1,24 +1,41 @@
+import { AuthApi, LoginDto, Role } from '@fullstack/sdk';
 // eslint-disable-next-line import/no-extraneous-dependencies
 import { EMAIL as EMAIL_CONST } from '@fullstack/server/src/auth/shared/Constants';
 import { Button, Checkbox, FormControlLabel, TextField } from '@mui/material';
-import { useState } from 'react';
+import { useContext, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { apiConfiguration } from '../../config';
+import { AuthContext } from '../../contexts';
+
+interface Error {
+  status: number;
+}
+
+type LocationProps = {
+  state: {
+    from: Location;
+  };
+};
+
+const LoginForm = ({ userRole }: { userRole: Role }) => {
+  const navigate = useNavigate();
+  const location = useLocation() as unknown as LocationProps;
```

Comment:
```suggestion
  const location = useLocation() as Location & LocationProps;
```

Chyba dodatkowy `as unknown` nie jest potrzebny przy takim overlapie typów? Plus można dodać część wspólną z `Location` (z `react-router-dom`, nie DOMu), żeby nie tracić przy okazji przy castowaniu innych pól z `location` kosztem dobrej typizacji `state`.